From 07935ccc94dc85fc798770eff8ac664d159aae9e Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:13 -0400
Subject: [tools] perf: Fix double free in perf test 21 (code-reading.c)

Message-id: <1410181752-25631-129-git-send-email-jolsa@redhat.com>
Patchwork-id: 91982
O-Subject: [PATCH RHEL7.1 BZ1134356 128/307] perf tools: Fix double free in perf test 21 (code-reading.c)
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit ae450a7d054cd1fe8d003a085858ca5b90cc74d9
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Thu Apr 10 12:02:54 2014 +0300

description
===========
perf_evlist__delete() deletes attached cpu and thread maps
but the test is still using them, so remove them from the
evlist before deleting it.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/code-reading.c b/tools/perf/tests/code-reading.c
index 653a8fe..bfb1869 100644
--- a/tools/perf/tests/code-reading.c
+++ b/tools/perf/tests/code-reading.c
@@ -504,6 +504,7 @@ static int do_test_code_reading(bool try_kcore)
   if (ret < 0) {
    if (!excl_kernel) {
     excl_kernel = true;
+    perf_evlist__set_maps(evlist, NULL, NULL);
     perf_evlist__delete(evlist);
     evlist = NULL;
     continue;
-- 
1.7.1